java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java

```diff
@@ -120,20 +120,17 @@ public void chainOptions() {
 
     @Override
     public synchronized Rewriter apply(org.kframework.definition.Definition def) {
-        Module mainModule = def.mainModule();
-        TermContext initializingContext = TermContext.builder(new GlobalContext(fs, globalOptions, krunOptions,
+        GlobalContext initializingContext = new GlobalContext(fs, globalOptions, krunOptions,
                 kproveOptions, javaExecutionOptions, kem, smtOptions, hookProvider, files, Stage.INITIALIZING, profiler,
-                kprint, def))
-                .freshCounter(0).build();
-        Definition definition;
-        definition = initializeDefinition.invoke(mainModule, kem, initializingContext.global());
+                kprint, def);
+        Definition definition = initializeDefinition.invoke(def.mainModule(), kem, initializingContext);
         GlobalContext rewritingContext = new GlobalContext(fs, globalOptions, krunOptions,
                 kproveOptions, javaExecutionOptions, kem, smtOptions, hookProvider, files, Stage.REWRITING, profiler,
                 kprint, def);
         rewritingContext.setDefinition(definition);
 
-        return new SymbolicRewriterGlue(mainModule, definition, definition, transitions,
-                initializingContext.getCounterValue(), rewritingContext, kem, files, kompileOptions, sw);
+        return new SymbolicRewriterGlue(def.mainModule(), definition, definition, transitions,
+                0, rewritingContext, kem, files, kompileOptions, sw);
```

Comment:
I do remember there was an issue that was fixed by having `initializingContext.getCounterValue()` instead of `0`, but I cannot recall the original issue.

Reply:
I guess that if we have more than one spec rules, this initial counter value needs to be different for each spec proof.  This counter is used to handle `?X`, if I remember correctly.

Reply:
Look at my comment above.